### DISCUSSION TITLE
How to handle jsx?

### DISCUSSION THREAD

**Discussion by pmaier983** [i18next/i18next#1535] [Q&A] [answered]
I am building the scaffolding for a conversion from gatsby -> nextjs for a documentation site.

The old gatsby code has almost all of its translations in one large tsx file, where translations are interspersed with jsx such as ```<Buttons/>, <string/>, <code/>``` etc.

Example of the gatsby code `key:value` => `defaultValues: jsxFunction`
```ts
defaultValues: (goToSection) => (
  <>
        <p>
          <b className={typographyStyles.note}>Note:</b> It is encouraged that
          you set a <code>defaultValue</code> for all inputs to non-
          <code>undefined</code> such as the empty string or <code>null</code>.
        </p>
          <button
            className={buttonStyles.codeAsLink}
            onClick={() => goToSection("Controller")}
          >
            Controller
          </button>{" "}
          component via its <code>defaultValue</code> property.
        </p>
  <>
```

1. Do I need to convert this all to strings that can be stored in a json file and use interpolation?
2. How much will I be able to use the trans-component? https://react.i18next.com/legacy-v9/trans-component

Thank you!
Phillip

**Comment by jamuhl** [comment, marked as answer]
1. / 2.) yes...and yes

There is no magic. Depending on situation you can use `<div>{t(key)}</div>` or `Trans`

Samples: https://github.com/i18next/react-i18next/tree/master/example
Docs: https://react.i18next.com/

react-i18next: https://github.com/i18next/react-i18next
next-i18next: https://github.com/isaachinman/next-i18next

**Reply by pmaier983** [reply]
Thank you for helping me! I’ve got a long road ahead of me.
